PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -162,32 +162,35 @@ function transformReports(inputReportFiles: string[], index: number)
                 reportType = ReportType.XML_XUNIT;
 
             } else if (node.name == 'CodingStandards') {
-                bStaticAnalysisResult = true;
+                bParsingStaticAnalysisResult = true;
 
             } else if (node.name == 'Rule') {
                 let ruleId = node.attributes.id;
                 let analyzerId = node.attributes.analyzer;
                 if (!bLegacyReport) {
-                    if(!ruleAnalyzerPairs.get(ruleId) && isDtpSettingsValid) {
-                        promisesOfGettingRuleDocUrl.push(getRulesDoc(ruleId, analyzerId));
+                    // A <Rule> has a rule ID and analyzer ID in a non-legacy report
+                    if (isDtpSettingsValid) {
+                        ruleDocUrlPromises.push(getRuleDoc(ruleId, analyzerId));
                     }
-                    ruleAnalyzerPairs.set(ruleId, analyzerId);
+                    ruleAnalyzerMap.set(ruleId, analyzerId);
                 } else if (node.attributes.cat == 'GLOBAL') {
-                    ruleIdsWithCatAsGlobal.add(ruleId);
+                    rulesInGlobalCategory.add(ruleId);
                 }
 
-            } else if (bStaticAnalysisResult && bLegacyReport && node.name.endsWith('Viol')) {
+            } else if (bParsingStaticAnalysisResult && bLegacyReport && node.name.endsWith('Viol')) {
                 let ruleId = node.attributes.rule;
-                let analyzerId = mapToAnalyzer(ruleId, node.name);
-                if(!ruleAnalyzerPairs.get(ruleId) && isDtpSettingsValid) {
-                    promisesOfGettingRuleDocUrl.push(getRulesDoc(ruleId, analyzerId));
+                if(!ruleAnalyzerMap.has(ruleId)) {
```

Comment:
1. Map.has() is more efficient than Map.get()
2. We only need to update the map when the rule doesn't already exist."